website/docs/cloud-docs/workspaces/settings/vcs.mdx

```diff
@@ -4,82 +4,63 @@ page_title: VCS Connections - Workspaces - Terraform Cloud and Terraform Enterpr
-- **Mark the entire repository as relevant.** If you set the "Automatic Run Triggering" setting to "Always Trigger Runs," Terraform Cloud will assume that anything in the repository might affect the workspace's configuration, and will queue runs for any change.
+#### Only trigger runs when files in specified paths change
 
-  This can be useful for repos that don't have multiple configurations but require a working directory for some other reason. It's usually not what you want for true monorepos, since it queues unnecessary runs and slows down your ability to provision infrastructure.
+This option instructs Terraform to begin new runs only for changes that affected specified files and directories. This behavior also applies to [speculative plans](/cloud-docs/run/#speculative-plans) on pull requests.
 
-### Error Handling for Run Triggers
+You can use trigger patterns and trigger prefixes in the **Add path** field to specify groups of files and directories.
+- **Trigger Patterns:** (Recommended) Use glob patterns to specify the files that should trigger a new run. For example, `/submodule/**/*.tf`, specifies all files with the `.tf` extension that are nested below the `submodule` directory. You can also use more complex patterns like `/**/networking/**/*`, which specifies all files in the `networking` folder at this level of the file path (e.g., `/submodule/service-1/networking/private/main.tf`). Refer to [TODO: create glob patterns page](link) for details.
+- ** Trigger Prefixes:** Terraform Cloud will queue runs for changes in any of the specified trigger directories matching the provided prefixes (including the working directory). For example, if you use a top-level `modules` directory to share Terraform code across multiple configurations, changes to the shared modules are relevant to every workspace that uses that repo. You can add `modules` as a trigger directory for each workspace to make sure they notice any changes to shared code.
 
-Terraform Cloud retrieves the changed files for each push or pull request using your VCS provider's API. If for some reason the list of changed files cannot be retrieved, or if it is too large to process, the default behavior is to trigger runs on all attached workspaces. Should this happen, you may see several runs with state "Planned", due to the push resulting in no changes to infrastructure.
+-> **Note:** Terraform triggers runs on all attached workspaces if it does not receive a list of changed files or if that list is too large to process. When this happens, Terraform Cloud may show several runs with completed plans that do not result in infrastructure changes.
 
-## Automatic Speculative Plans
+### Terraform Working Directory
```

Comment:
@matejrisek I'm confused by this. Can we set this on this page, or do we only set it on "General Settings"?  

Regardless, the old docs had a note about how run triggering was only relevant if you hadn't specified a working directory. Should we bring back that note? If not, we can remove this content entirely. 

Here's the note for reference: 
https://www.terraform.io/cloud-docs/workspaces/settings/vcs#automatic-run-triggering

"Note: This setting only affects workspaces that specify a Terraform working directory. Without a working directory, the entire repository is considered relevant and any change will trigger a run. You can set [The working directory](https://www.terraform.io/cloud-docs/workspaces/settings#terraform-working-directory) on the **General** settings page."

Reply:
It's not true anymore, with the introduction of `trigger-patterns` we're removing the dependency between working directory and triggers. Although if we choose `prefixes` as a type of triggers we want to use - they will still take working directory into consideration when deciding whether to trigger a run.

@omarismail and @aaabdelgany could you confirm on this? I think I've read in one of the recent conversations that we've decided to remove working directory as a prerequisite to input both `trigger-patterns` and `trigger-prefixes` in the UI.

P.S.
I think we've forgotten to update documentation on where we can set working directory - right now one can set it both through the VCS section and the General settings page.

Reply:
That is correct. A working directory is no longer necessary to add trigger patterns, but it is necessary to add trigger prefixes. We will flash a warning in the UI if the working directory is empty and they try to use trigger prefixes.

Reply:
ahhh okay! then I'm going to leave this in, and @matejrisek can draft something for it after this PR is merged.